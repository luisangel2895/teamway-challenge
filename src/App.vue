<template lang="pug">
#app
  Loader(v-if="loading")
  main
    router-view
  MobileNav
</template>

<script lang="ts">
import { Options, Vue } from 'vue-class-component';
import Loader from '@/views/Loader.vue';
import MobileNav from '@/components/shared/MobileNav.vue';

@Options({
  components: {
    Loader,
    MobileNav,
  },
})
export default class App extends Vue {
  loading = true;

  async mounted(): Promise<void> {
    await setTimeout(() => {
      this.loading = false;
    }, 3000);
    // this.getNews();
  }
}
</script>

<style lang="scss">
#app {
  main {
    height: calc(100vh - 80px);
  }
}

// @media (min-width: 450px) {
//   #app {
//     width: 450px !important;
//   }
// }
</style>
