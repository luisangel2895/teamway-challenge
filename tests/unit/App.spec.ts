import { shallowMount } from '@vue/test-utils';
// Components
import App from '@/App.vue';

const wrapper = shallowMount(App, {
  global: {
    stubs: ['router-view'],
  },
});

describe('Test in component App.vue', (): void => {
  it('Match with snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('Components (loader, main, router-view, mobile-nav) are mounted correctly', (): void => {
    expect(wrapper.find('loader-stub').exists()).toBeTruthy();
    expect(wrapper.find('main').exists()).toBeTruthy();
    expect(wrapper.find('router-view-stub').exists()).toBeTruthy();
    expect(wrapper.find('mobile-nav-stub').exists()).toBeTruthy();
  });
  it('Loading start in the mounted and the duration is 3s', async (): Promise<void> => {
    expect(wrapper.vm.loading).toBe(true);
    await new Promise((resolve) => setTimeout(resolve, 3000));
    expect(wrapper.vm.loading).toBe(false);
  });
});
