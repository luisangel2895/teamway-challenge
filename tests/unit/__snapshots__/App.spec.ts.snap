// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in component App.vue Match with snapshot 1`] = `
<div id="app">
  <loader-stub></loader-stub>
  <main>
    <router-view-stub></router-view-stub>
  </main>
  <mobile-nav-stub></mobile-nav-stub>
</div>
`;
